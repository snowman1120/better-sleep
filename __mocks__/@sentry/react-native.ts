jest.mock('@sentry/react-native', () => ({
  setTagsContext: jest.fn(),
  setExtraContext: jest.fn(),
  captureBreadcrumb: jest.fn()
}))
