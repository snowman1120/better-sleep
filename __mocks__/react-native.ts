jest.mock('react-native', () => ({
  StyleSheet: {
    hairlineWidth: 1,
    create: () => ({}),
    flatten(arr: any) {
      return arr.reduce((res: any, item: any) => Object.assign(res, item), {})
    }
  },
  Platform: {
    OS: jest.fn(() => 'android'),
    version: jest.fn(() => 25)
  },
  Dimensions: {
    get: () => {
      return { width: 100, height: 200 }
    }
  },
  I18nManager: {
    isRTL: false
  },
  NativeModules: {
    RNDocumentPicker: () => {},
    RNSentry: () => jest.fn()
  },

  Easing: {
    bezier: () => {}
  },
  View: () => 'View',
  ViewPropTypes: {
    propTypes: {
      style: {}
    }
  },
  Text: () => 'Text',
  TouchableNativeFeedback: () => 'TouchableNativeFeedback',
  TouchableOpacity: () => 'TouchableOpacity',

  TouchableWithoutFeedback: () => 'TouchableWithoutFeedback',
  Animated: {
    View: () => 'Animated.View',
    interpolate: jest.fn(),
    Value: jest.fn().mockImplementation(() => {
      return { interpolate: jest.fn() }
    })
  }
}))
