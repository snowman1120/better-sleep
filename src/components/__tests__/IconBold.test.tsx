import * as React from 'react'
import 'react-native'
import * as renderer from 'react-test-renderer'
import { IconBold } from '../iconRegular'

it('IconBold Renders correctly', () => {
  const rendered = renderer
    .create(
      <IconBold height={20} width={20} name="taskListEdit" fill="black" />
    )
    .toJSON()
  expect(rendered).toMatchSnapshot()
})
