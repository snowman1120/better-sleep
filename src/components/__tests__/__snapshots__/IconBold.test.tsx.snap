// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconBold Renders correctly 1`] = `
<Svg
  height="20"
  viewBox="0 0 24 24"
  width="20"
>
  <G
    fill="black"
  >
    <Path
      d="M17,21a1,1,0,0,1-1,1H3a1,1,0,0,1-1-1V3A1,1,0,0,1,3,2H4.75a.25.25,0,0,1,.25.25V3.5A1.5,1.5,0,0,0,6.5,5h6A1.5,1.5,0,0,0,14,3.5V2.249A.25.25,0,0,1,14.25,2H16a1,1,0,0,1,1,1V8.314a.25.25,0,0,0,.427.177l1.5-1.5A.249.249,0,0,0,19,6.814V3a3,3,0,0,0-3-3H3A3,3,0,0,0,0,3V21a3,3,0,0,0,3,3H16a3,3,0,0,0,3-3V15.764a.25.25,0,0,0-.427-.177l-1.5,1.5a.249.249,0,0,0-.073.177Z"
    />
    <Path
      d="M8.469,6.663a.752.752,0,0,0-1.055.118l-1.3,1.627a.248.248,0,0,1-.181.094.251.251,0,0,1-.191-.073l-.71-.71A.75.75,0,1,0,3.97,8.78l1.5,1.5a.759.759,0,0,0,1.116-.061h0l2-2.5A.751.751,0,0,0,8.469,6.663Z"
    />
    <Path
      d="M8.469,12.163a.752.752,0,0,0-1.055.118l-1.3,1.627A.248.248,0,0,1,5.931,14a.251.251,0,0,1-.191-.073l-.71-.71A.75.75,0,1,0,3.97,14.28l1.5,1.5a.759.759,0,0,0,1.116-.061h0l2-2.5A.751.751,0,0,0,8.469,12.163Z"
    />
    <Path
      d="M14.25,8.5h-3.5a.75.75,0,0,0,0,1.5h3.5a.75.75,0,0,0,0-1.5Z"
    />
    <Path
      d="M23.586,8.453a1.414,1.414,0,1,0-2-2L11.573,16.466a.244.244,0,0,0-.06.1l-.665,2a.5.5,0,0,0,.632.632l2-.665a.244.244,0,0,0,.1-.06Z"
    />
  </G>
</Svg>
`;
